Assets/Scripts/Models/CharacterNameManager.cs

```diff
@@ -0,0 +1,69 @@
+﻿#region License
+// ====================================================
+// Project Porcupine Copyright(C) 2016 Team Porcupine
+// This program comes with ABSOLUTELY NO WARRANTY; This is free software,
+// and you are welcome to redistribute it under certain conditions; See
+// file LICENSE, which is part of this source code package, for details.
+// ====================================================
+#endregion
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// Character name manager that holds all the possible names and tryes to give a random and unused name everytime is requested.
+/// </summary>
+public class CharacterNameManager
+{
+    private static Queue<string> unusedNames;
+    private static Queue<string> usedNames;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="CharacterNameManager"/> class.
+    /// </summary>
+    public CharacterNameManager()
+    {
+        unusedNames = new Queue<string>();
+        usedNames = new Queue<string>();
+    }
+
+    /// <summary>
+    /// Load new names to use for Characters and add them in a random way.
+    /// </summary>
+    /// <param name="nameStrings">An array of name strings.</param>
+    public static void LoadNames(string[] nameStrings)
+    {
+        // Randomize the given strings
+        for (int i = 0; i < nameStrings.Length; i++)
+        {
+            int index = Random.Range(0, nameStrings.Length - 1);
+            string tempName = nameStrings[index];
+            nameStrings[index] = nameStrings[i];
+            nameStrings[i] = tempName;
+        }
+
+        // Add all the names to the unused queue in the random order
+        foreach (string name in nameStrings)
+        {
+            unusedNames.Enqueue(name);
+        }
+    }
+
+    /// <summary>
+    /// Gets a name which wasn't used before.
+    /// </summary>
+    /// <returns>Psossibly an unused name.</returns>
```

Comment:
Possibly -- also this conflicts a bit with the summary. Could even drop the returns tag... I would suggest something like
```c#
/// <summary>
/// Returns a randomly chosen name, prioritizing names which have not been used yet.
/// </summary>
```